snapcraft/internal/repo/apt_cache.py

```diff
@@ -140,6 +140,23 @@ def _verify_marked_install(self, package: apt.Package):
     def is_package_valid(self, package_name: str) -> bool:
         return package_name in self.cache or self.cache.is_virtual_package(package_name)
 
+    def is_package_installed(self, package_name: str, package_version: str) -> bool:
```

Comment:
it took me a minute to understand what was going on here... there still is a `is_package_installed()` in _deb.

Maybe `is_package_version_installed`?  still feels weird to have the function do the comparison.

I'd suggest instead adding a new parameter `resolve_virtual_packages: bool` to `get_installed_version(...)` and pass it along to `_deb.is_package_installed(...)`

Reply:
I missed the one in ._deb, that removes my entire move. I'll move the logic back and add an extra parameter to recurse